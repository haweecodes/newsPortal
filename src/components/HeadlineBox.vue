<template>
  <v-card width="100%" height="100%">
    <v-card-title>
      <slot name="HeadlineTitle"></slot>
    </v-card-title>
    <v-card-text>
      <slot name="HeadlineContent"></slot>
    </v-card-text>
    <v-card-actions>
      <slot name="HeadlineAction"></slot>
    </v-card-actions>
  </v-card>
</template>

<script>
export default {
  name: "HeadlineBox",
};
</script>

<style></style>
