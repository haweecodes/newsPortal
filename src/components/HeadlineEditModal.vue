<template>
  <v-dialog v-model="open" width="500">
    <v-card>
      <v-card-title>Edit headline title</v-card-title>
      <v-card-text class="mt-2">
        <v-text-field
          v-model="newHeadlineTitle"
          label="New headline title"
          :rules="[
            () => !!newHeadlineTitle || 'This field is required',
            () =>
              (!!newHeadlineTitle && newHeadlineTitle.length < 256) ||
              'Headline must be less than 255 characters',
          ]"
          counter="255"
          outlined
        ></v-text-field>
      </v-card-text>

      <v-card-actions>
        <v-spacer></v-spacer>

        <v-btn color="red darken-1" text @click="closeModal"> Cancel </v-btn>

        <v-btn
          color="green darken-1"
          text
          @click="editHeadlineTitle"
          :disabled="checkHeadlineLength"
        >
          Confirm
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
export default {
  name: "HeadlineEditModal",
  props: ["open", "onHeadlineChange", "onModalClose"],
  data: () => ({
    newHeadlineTitle: "",
  }),
  computed: {
    checkHeadlineLength() {
      return this.newHeadlineTitle.length > 255;
    },
  },
  methods: {
    editHeadlineTitle() {
      this.$emit("onHeadlineChange", this.newHeadlineTitle);
    },
    closeModal() {
      this.$emit("onModalClose");
    },
  },
};
</script>

<style></style>
