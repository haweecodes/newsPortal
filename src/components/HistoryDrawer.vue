<template>
  <v-navigation-drawer v-model="show" absolute temporary>
    <v-list-item>
      <v-list-item-content>
        <v-list-item-title>History</v-list-item-title>
      </v-list-item-content>
    </v-list-item>

    <v-divider></v-divider>

    <v-list dense v-if="historyList.length > 0">
      <v-list-item v-for="history in historyList" :key="`${history.time}`">
        <v-list-item-content>
          <v-list-item-subtitle>{{ history.time | transformDate }}</v-list-item-subtitle>
          <v-list-item-title>{{ history.title }}</v-list-item-title>
        </v-list-item-content>
      </v-list-item>
    </v-list>
    <p v-else>No history to show</p>
  </v-navigation-drawer>
</template>

<script>
import { getHistoryFromLocalStorage } from '../utils/helper';

export default {
  name: 'HistoryDrawer',
  model: {
    prop: 'drawer',
    event: 'change',
  },
  props: ['drawer'],
  data: () => ({
    historyList: [],
  }),
  computed: {
    show: {
      get() {
        return this.drawer;
      },
      set(value) {
        this.$emit('change', value);
      },
    },
  },
  beforeMount() {
    this.historyList = getHistoryFromLocalStorage();
  },
};
</script>

<style></style>
