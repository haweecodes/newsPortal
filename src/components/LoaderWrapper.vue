<template>
  <section v-if="!loading">
    <slot name="content" />
  </section>
  <section v-else class="d-flex justify-center align-center" >
    <v-progress-circular
      :size="70"
      :width="7"
      indeterminate
      color="primary"
    ></v-progress-circular>
    <!-- <v-row>
      <v-col v-for="i in 10" :key="i" cols="3">
        <v-skeleton-loader
          class="mx-auto"
          max-width="300"
          type="card"
        ></v-skeleton-loader>
      </v-col>
    </v-row> -->
  </section>
</template>

<script>
export default {
  name: 'LoaderWrapper',
  props: ['loading'],
};
</script>

<style></style>
