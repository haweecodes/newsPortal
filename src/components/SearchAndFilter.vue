<template>
  <v-row class="ma-2" wrap>
    <v-col md="4" sm="8" xs="12" offset-md="6" offset-sm="0">
      <v-text-field
        placeholder="Press enter to search..."
        outlined
        hide-details
        clearable
        @click:clear="clearHeadlineSearch"
        v-model="searchText"
        @keyup="headlineSearch"
      ></v-text-field
    ></v-col>
    <v-col md="2" sm="4" xs="12" class="d-flex justify-end align-center">
      <v-menu offset-y max-height="400">
        <template v-slot:activator="{ on, attrs }">
          <v-btn width="auto" color="primary" dark v-bind="attrs" v-on="on">
            Filter
          </v-btn>
        </template>
        <v-list>
          <v-list-item
            v-for="source in getSourceList"
            :key="source.id"
            @click.stop="setFilterSource(source.id)"
          >
            <v-list-item-title>{{ source.name }}</v-list-item-title>
          </v-list-item>
        </v-list>
      </v-menu>
      <v-btn width="auto" class="ml-2" @click.stop="setFilterSource(null)">Reset</v-btn>
    </v-col>
  </v-row>
</template>

<script>
import { throttleFunction } from '../utils/helper';

export default {
  name: 'SearchAndFilter',
  data: () => ({
    searchText: '',
  }),
  computed: {
    getSourceList() {
      return this.$store.state.sourceList;
    },
  },
  beforeMount() {
    this.$store.dispatch('fetchSourceList');
  },

  methods: {
    headlineSearch() {
      if (this.searchText !== '' && this.searchText !== null) {
        throttleFunction(this.callSearchApi, 600);
      }
    },
    callSearchApi() {
      this.$store.dispatch('fetchSearchedHeadline', this.searchText);
    },
    clearHeadlineSearch() {
      this.$store.dispatch('fetchHeadlineList');
    },
    setFilterSource(value) {
      this.$store.commit('setFilterSource', value);
    },
  },
};
</script>
