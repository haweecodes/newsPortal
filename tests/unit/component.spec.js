import Vuetify from 'vuetify';
import {
  mount,
  shallowMount,
  createLocalVue,
} from '@vue/test-utils';
import Vuex from 'vuex';
import HeadlineBox from '../../src/components/HeadlineBox.vue';
import HeadlineEditModal from '../../src/components/HeadlineEditModal.vue';
import HistoryDrawer from '../../src/components/HistoryDrawer.vue';
import LoaderWrapper from '../../src/components/LoaderWrapper.vue';
import SearchAndFilter from '../../src/components/SearchAndFilter.vue';

const vuetify = new Vuetify();
const localVue = createLocalVue();

describe('test HeadlineBox', () => {
  it('test headlinebox mount', () => {
    const wrapper = mount(HeadlineBox, {
      slots: {
        HeadlineTitle: '<div>title</div>',
        HeadlineContent: '<div>content</div>',
        HeadlineAction: '<div>action</div>',
      },
    });

    expect(wrapper.html()).toContain('<div>title</div>');
    expect(wrapper.html()).toContain('<div>content</div>');
    expect(wrapper.html()).toContain('<div>action</div>');
  });
});

describe('test HeadlineEditModal', () => {
  it('test HeadlineEditModal mount', () => {
    const wrapper = shallowMount(HeadlineEditModal, {
      props: {
        open: true,
        onHeadlineChange: () => {},
        onModalClose: () => {},
      },
      mocks: {
        $vuetify: {
          breakpoint: {},
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe('test HistoryDrawer', () => {
  it('test HistoryDrawer mount', () => {
    const wrapper = shallowMount(HistoryDrawer, {
      props: {
        drawer: true,
      },
      mocks: {
        $vuetify: {
          breakpoint: {},
        },
      },
      vuetify,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe('test LoaderWrapper', () => {
  it('test LoaderWrapper mount', () => {
    const wrapper = shallowMount(LoaderWrapper, {
      props: {
        loading: true,
      },
      mocks: {
        $vuetify: {
          breakpoint: {},
        },
      },
      vuetify,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe('test SearchAndFilter', () => {
  let actions;
  let state;
  let store;

  beforeEach(() => {
    localVue.use(Vuex);

    state = {
      sourceList: [],
    };
    actions = {
      fetchSourceList: jest.fn(),
    };
    store = new Vuex.Store({
      actions,
      state,
    });
  });
  it('test SearchAndFilter mount', () => {
    const wrapper = shallowMount(SearchAndFilter, {
      localVue,
      store,
      mocks: {
        $vuetify: {
          breakpoint: {},
        },
      },
      vuetify,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
