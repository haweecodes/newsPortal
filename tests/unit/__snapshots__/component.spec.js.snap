// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HeadlineEditModal test HeadlineEditModal mount 1`] = `
<v-dialog-stub contentclass="" opendelay="0" closedelay="0" openonclick="true" origin="center center" retainfocus="true" transition="dialog-transition" width="500">
  <v-card-stub loaderheight="4" tag="div">
    <v-card-title-stub tag="div">Edit headline title</v-card-title-stub>
    <v-card-text-stub tag="div" class="mt-2">
      <v-text-field-stub errorcount="1" errormessages="" messages="" rules="[Function],[Function]" successmessages="" value="" backgroundcolor="" label="New headline title" loaderheight="2" clearicon="$clear" counter="255" outlined="true" type="text"></v-text-field-stub>
    </v-card-text-stub>
    <v-card-actions-stub tag="div">
      <v-spacer-stub tag="div"></v-spacer-stub>
      <v-btn-stub color="red darken-1" tag="button" activeclass="" text="true" type="button"> Cancel </v-btn-stub>
      <v-btn-stub color="green darken-1" tag="button" activeclass="" text="true" type="button">
        Confirm
      </v-btn-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-dialog-stub>
`;

exports[`test HistoryDrawer test HistoryDrawer mount 1`] = `
<v-navigation-drawer-stub absolute="true" mobilebreakpoint="1264" height="100%" minivariantwidth="56" src="" tag="aside" temporary="true" width="256">
  <v-list-item-stub activeclass="" tag="div">
    <v-list-item-content-stub tag="div">
      <v-list-item-title-stub tag="div">History</v-list-item-title-stub>
    </v-list-item-content-stub>
  </v-list-item-stub>
  <v-divider-stub></v-divider-stub>
  <p>No history to show</p>
</v-navigation-drawer-stub>
`;

exports[`test LoaderWrapper test LoaderWrapper mount 1`] = `<section></section>`;

exports[`test SearchAndFilter test SearchAndFilter mount 1`] = `
<v-row-stub tag="div" wrap="" class="ma-2">
  <v-col-stub sm="8" md="4" offsetsm="0" offsetmd="6" tag="div" xs="12">
    <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" hidedetails="true" loaderheight="2" clearable="true" clearicon="$clear" outlined="true" placeholder="Press enter to search..." type="text"></v-text-field-stub>
  </v-col-stub>
  <v-col-stub sm="4" md="2" tag="div" xs="12" class="d-flex justify-end align-center">
    <v-menu-stub opendelay="0" closedelay="0" openonclick="true" contentclass="" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" closeonclick="true" closeoncontentclick="true" maxheight="400" offsety="true" origin="top left" transition="v-menu-transition">
      <v-list-stub tag="div"></v-list-stub>
    </v-menu-stub>
    <v-btn-stub width="auto" tag="button" activeclass="" type="button" class="ml-2">Reset</v-btn-stub>
  </v-col-stub>
</v-row-stub>
`;
